docs/source/layers/layer_like.rst

```diff
@@ -0,0 +1,68 @@
+Layer-Like Objects
```

Comment:
@davidbrochart and @martinRenou, please review this explanation and feel free to modify. This is what I came up with on a whim to document the interface.